nodestream/databases/neo4j/query_executor.py

```diff
@@ -81,7 +82,16 @@ async def execute(self, query: Query, log_result: bool = False):
                 "uri": self.driver._pool.address.host,
             },
         )
-        await self.driver.verify_connectivity()
+
+        try:
+            await self.driver.verify_connectivity()
+        except ServiceUnavailable:
```

Comment:
Another interesting find is that when there is a time out while establishing a connection it seems that neo4j is supposed to throw a ClientError
https://github.com/neo4j/neo4j-python-driver/blob/e2448ec6e044f3aea8ee81ab6df99f37e61a2e36/src/neo4j/_async/io/_pool.py#L313

Reply:
```
Traceback (most recent call last):
  File ".venv/lib/python3.11/site-packages/neo4j/_async_compat/network/_bolt_socket.py", line 386, in connect
    s = await cls._connect_secure(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".venv/lib/python3.11/site-packages/neo4j/_async_compat/network/_bolt_socket.py", line 244, in _connect_secure
    raise ServiceUnavailable(
neo4j.exceptions.ServiceUnavailable: Timed out trying to establish connection to ResolvedIPv6Address(('::1', 7687, 0, 0))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "nodestream/nodestream/databases/neo4j/query_executor.py", line 87, in execute
    await self.driver.verify_connectivity()
  File ".venv/lib/python3.11/site-packages/neo4j/_async/driver.py", line 965, in verify_connectivity
    await self._get_server_info()
  File ".venv/lib/python3.11/site-packages/neo4j/_async/driver.py", line 1179, in _get_server_info
    return await session._get_server_info()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".venv/lib/python3.11/site-packages/neo4j/_async/work/session.py", line 177, in _get_server_info
    await self._connect(READ_ACCESS, liveness_check_timeout=0)
  File ".venv/lib/python3.11/site-packages/neo4j/_async/work/session.py", line 135, in _connect
    await super()._connect(
  File ".venv/lib/python3.11/site-packages/neo4j/_async/work/workspace.py", line 178, in _connect
    self._connection = await self._pool.acquire(**acquire_kwargs_)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".venv/lib/python3.11/site-packages/neo4j/_async/io/_pool.py", line 537, in acquire
    return await self._acquire(
           ^^^^^^^^^^^^^^^^^^^^
  File ".venv/lib/python3.11/site-packages/neo4j/_async/io/_pool.py", line 322, in _acquire
    return await connection_creator()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".venv/lib/python3.11/site-packages/neo4j/_async/io/_pool.py", line 174, in connection_creator
    connection = await self.opener(
                 ^^^^^^^^^^^^^^^^^^
  File ".venv/lib/python3.11/site-packages/neo4j/_async/io/_pool.py", line 511, in opener
    return await AsyncBolt.open(
           ^^^^^^^^^^^^^^^^^^^^^
  File ".venv/lib/python3.11/site-packages/neo4j/_async/io/_bolt.py", line 381, in open
    await AsyncBoltSocket.connect(
  File ".venv/lib/python3.11/site-packages/neo4j/_async_compat/network/_bolt_socket.py", line 424, in connect
    raise ServiceUnavailable(
neo4j.exceptions.ServiceUnavailable: Couldn't connect to localhost:7687 (resolved to ('[::1]:7687', '127.0.0.1:7687')):
Timed out trying to establish connection to ResolvedIPv6Address(('::1', 7687, 0, 0))
Timed out trying to establish connection to ResolvedIPv4Address(('127.0.0.1', 7687))
```

This was the stacktrace provided when the timeouts were happening. Maybe deeper in the driver source it was a different type but it was eventually rethrown as a ServiceUnavailable error when it got to the driver code.

Reply:
Make sense. Looks like there is a variety of undefined issues that can cause `ServiceUnavailable`.